x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalVerifier.java

```diff
@@ -47,6 +48,13 @@ public Collection<Failure> verify(PhysicalPlan plan) {
                         )
                     );
                 }
+            } else if (p instanceof QueryStringFilterExec queryStringFilterExec) {
+                failures.add(
```

Comment:
For now, there's no usage of MATCH directly on non-index based sources - we can implement that separately